codec-http2/src/main/java/io/netty/handler/codec/http2/CharSequenceMap.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http2;
+
+import io.netty.handler.codec.DefaultHeaders;
+import io.netty.handler.codec.UnsupportedValueConverter;
+import io.netty.handler.codec.ValueConverter;
+
+import static io.netty.util.AsciiString.CASE_INSENSITIVE_HASHER;
+import static io.netty.util.AsciiString.CASE_SENSITIVE_HASHER;
+
+final class CharSequenceMap<V> extends DefaultHeaders<CharSequence, V> {
```

Comment:
This is a bit of a hack to leverage the DefaultHeader's ability to take a `HashingStrategy`. We had some `Map` objects internally that depended on being able to compare `CharSequence` objects (mostly `String` and `AsciiString`) using `contentEquals*` instead of `equals`.